### ISSUE TITLE
ice shelf 2D test cases fail in debug mode

### ISSUE THREAD

**Issue by mark-petersen** [MPAS-Dev/compass#172]
These two test cases fail with gnu and intel debug. They pass with optimized:
```
ocean/ice_shelf_2d/5km/z-star/restart_test
  test execution:      ERROR
  see: case_outputs/ocean_ice_shelf_2d_5km_z-star_restart_test.log
ocean/ice_shelf_2d/5km/z-level/restart_test
  test execution:      ERROR
  see: case_outputs/ocean_ice_shelf_2d_5km_z-level_restart_test.log
```
Using compass f5dcaf98 and E3SM 0590a25151

**Comment by mark-petersen** [comment]
The logfile contains:
```
220 Backtrace for this error:
222 Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.
224 Backtrace for this error:
...
236     at /usr/projects/climate/mpeterse/repos/E3SM/master/components/mpas-ocean/src/analysis_members/mpas_ocn_global_stats.f90:1132
237 #2  0xaa0c29 in ocn_compute_analysis_members
238     at /usr/projects/climate/mpeterse/repos/E3SM/master/components/mpas-ocean/src/analysis_members/mpas_ocn_analysis_driver.f90:1195
```
which is here:
```
vi /usr/projects/climate/mpeterse/repos/E3SM/master/components/mpas-ocean/src/analysis_members/mpas_ocn_global_stats.f90 +1132
1129       ! normalVelocity
1130       variableIndex = variableIndex + 1
1131       averages(variableIndex) = sums(variableIndex)/volumeEdgeGlobal
1132       rms(variableIndex) = sqrt(sumSquares(variableIndex)/volumeEdgeGlobal)
```
might be a nan in the volume edge array.

**Comment by xylar** [comment]
Thanks @mark-petersen.

> might be a nan in the volume edge array.

Yeah, or a zero in `volumeEdgeGlobal`.  Though maybe division by zero is a separate error?

**Comment by xylar** [comment]
I was able to reproduce the error on Chrysalis with intel and impi in debug mode.  I printed both terms in the `sqrt` and they're pretty bonkers, and resulting in a negative number.  So I think there's some important debugging to be done of the global stats AM in general with ice-shelf cavities and partial top cells.  Presumably, some invalid values that were previously zeroed out but now contain fill values are getting used in the calculations.
